src/common/Utilities.cpp

```diff
@@ -36,12 +36,12 @@ void convertFloatToPcm16(const float *source, int16_t *destination, int32_t numS
         float fval = source[i];
         fval += 1.0; // to avoid discontinuity at 0.0 caused by truncation
         fval *= 32768.0f;
-        auto sample = (int32_t) fval;
+        auto sample = static_cast<int32_t>(fval + 0.5f);
```

Comment:
This changes the behavior of the code from floor to round! Please remove the "+ 0.5f". 
Maybe it should match the similar code in:
"frameworks/av/media/libaaudio/src/utility/AAudioUtilities.cpp"
If so then, let's do that in a separate Pull Request.

Reply:
yeah, it is: so please check whether this helps or not ( rounding up or cut-off ). Let me restore it now

Reply:
done reverting, thanks